core/types/permissions_cache.go

```diff
@@ -617,3 +668,58 @@ func ValidateNodeForTxn(hexnodeId string, from common.Address) bool {
 
 	return false
 }
+
+func IsEEAPermission() bool {
+	return PermissionModel == EEA
+}
+
+//  checks if the account permission allows the transaction to be executed
+func IsTransactionAllowed(from common.Address, to common.Address, value *big.Int, gasPrice *big.Int, gasLimit *big.Int, payload []byte, transactionType TransactionType) error {
+	//if we have not reached QIP714 block return full access
+	if !PermissionsEnabled() {
+		return nil
+	}
+
+	switch PermissionModel {
+	case Default:
+		return nil
+
+	case Basic:
+		return isTransactionAllowedBasic(from, transactionType)
+
+	case EEA:
+		if PermissionTransactionAllowedFunc == nil {
+			return ErrNoPermissionForTxn
+		}
+
+		allowed, err := PermissionTransactionAllowedFunc(from, to, value, gasPrice, gasLimit, payload)
+		if err != nil {
```

Comment:
The actual `err` value is being lost here which may make troubleshooting difficult. I'd suggest logging the error message or including it within the returned error.

Reply:
done